python: added iio_readdev.py and iio_writedev.py

cristi-iacob opened the pull request:
Implemented the iio_readdev and iio_writedev modules with Python bindings. These modules can be used as examples for sending and receiving buffers.

rgetz commented:
Since this will be an example - can we add some sphinx comments, so when we merge #393 things show up?

tfcollins commented on bindings/python/examples/iio_writedev.py:
```diff
@@ -0,0 +1,221 @@
+    raw_arguments = sys.argv[1:]
+
+    try:
+        arguments, values = getopt.getopt(raw_arguments, unixOptions, gnuOptions)
```
Why was argparse not used? It much much simpler and in the core library.

cristi-iacob replied:
I didn't know argparse was in the core library; getopt just seemed simpler at the first glance for me. I have changed them to use argparse.

tfcollins commented on bindings/python/examples/iio_writedev.py:
```diff
@@ -0,0 +1,221 @@
+
+        while bytes_to_read > 0:
+            data = sys.stdin.buffer.read(bytes_to_read)
+            bytes_to_read -= len(data)
```
Won't this loop simply overwrite data when bytes_to_read is not 0 after the first loop? Therefore the first array is never passed to the write buffer.

cristi-iacob replied:
Yes, you are right.  I have modified; I think this modification should solve the problem.

tfcollins commented on bindings/python/examples/iio_writedev.py:
```diff
@@ -0,0 +1,221 @@
+
+        if ret == 0:
+            if app_running:
+                print('Unable to push buffer!')
```
Should this be an exception?

tfcollins commented on bindings/python/examples/iio_readdev.py:
```diff
@@ -0,0 +1,197 @@
+    buffer = iio.Buffer(dev, buffer_size)
+
+    if buffer is None:
+        print('Unable to create buffer!')
```
Should this be an exception?

rgetz commented:
Also - until we have it fixed, check out: https://app.codacy.com/manual/commodo/libiio/pullRequest?prid=5299649#newIssuesView

rgetz commented:
There still doesn't seem to be any markdown in the code that will help sphinx? Should there be?

cristi-iacob commented:
I have added some documentation for the functions from iio_readdev.py and iio_writedev.py.
Also I have added the .rst files for them, in order to be compatible with sphinx. If these files shouldn't have been added, I will delete the last commit.

tfcollins commented:
Can you add references to the index file to the examples? Otherwise I don't think you can get to them.

cristi-iacob commented:
I have added a new component on the main page, Tools. It contains references to iio_readdev and iio_writedev documentation pages.

rgetz commented:
Thanks - looks great.
